src/machine.rs

```diff
@@ -118,6 +122,17 @@ impl MemoryExtra {
                     .extern_statics
                     .insert(Symbol::intern("__cxa_thread_atexit_impl"), place.ptr.assert_ptr().alloc_id)
                     .unwrap_none();
+
+                // "environ"
```

Comment:
You are only doing this if the target is `linux`, which is why macOS tests are failing.

Reply:
Hmm it seems there's something else going on. From apple's manual: 


     Shared libraries and bundles don't have direct access to environ, which is only available to the loader ld(1) when a complete program is
     being linked.  The environment routines can still be used, but if direct access to environ is needed, the _NSGetEnviron() routine, defined in
     <crt_externs.h>, can be used to retrieve the address of environ at runtime.

Reply:
> From apple's manual:

Ah so macOS does not even use the extern static then.
But then we probably shouldn't provide the static on macOS either. Instead you can initialize the env vars first in `create_ecx`, and then only on linux also make the environment available as extern static.